map-demo.tf

```diff
@@ -0,0 +1,20 @@
+module "map-demo" {
+  source = "./modules/service"
+
+  name      = "map-demo"
+  subdomain = "map-demo"
+  image     = "smartatransit/map-demo:latest"
+  port      = 80
+
+  traefik_network_name = docker_network.traefik.id
+
+  //nginx publishes to port 80 on the host by default - to avoid a
```

Comment:
If the module is set up correctly (🙏 🙏) I think you want to delete the `endpoint_spec` here and instead change `port = 80` to `port = 4000` above.

Explanation that you don't need to read: this is just there because (for some frustrating reason) the nginx image is designed to try to grab port 80 on the _host_, which is bad because (1) traefik needs that port so there'll be a conflict, and (2) we don't want service containers to own host ports, we just want traefik to contact them internally. There doesn't seem to be any way to tell nginx to **not **grab a host port, but we can at least tell it to grab one other than port 80 so that it won't conflict with traefik and crash.

Your image is based on `node:10` which doesn't try to grab a host port in the first place, it just gives you the option of exposing the container in whichever way you feel is appropriate. All you need to do is set the `port` variable to 4000, which will in turn populate the `traefik.service.map-demo.port` label, which tells traefik that when it forwards a request to containers for this service, it should contact them on port 4000.